### PR TITLE
[@types/algoliasearch] Add proper types for ApiKey

### PR DESCRIPTION
**Pull request by bodinsamuel**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://www.algolia.com/doc/api-reference/api-methods/add-api-key/?language=javascript https://www.algolia.com/doc/api-reference/api-methods/list-api-keys/

--- 

## Changes
- Add proper types for ApiKey: `.addApiKey()` `.updateApiKey()` `.deleteApiKey()` `. listApiKeys()`
- Prettier got in the way (checking **Hide whitespace changes** will help)

### CODE REVIEW

File: types/algoliasearch/index.d.ts [hunk 1]
```diff
@@ -1356,8 +1375,41 @@ declare namespace algoliasearch {
...
-    status: 'published' | 'notPublished',
-    pendingTask: boolean,
+    status: 'published' | 'notPublished';
+    pendingTask: boolean;
+  }
+
+  interface ApiKey {
+    value: string;
+    createdAt: number;
+    acl: (
```
Comment (Haroenv) [inline types/algoliasearch/index.d.ts]: acl isn't given to every return right?

**Review by Haroenv — COMMENTED** [review]
looks good, quick question

File: types/algoliasearch/index.d.ts (hunk 1 above)
Comment (bodinsamuel) [reply types/algoliasearch/index.d.ts]: What do you mean? 
It's returned on GET /keys and /keys/{id} (the REST API doc is really not clear)

**Review by Haroenv — APPROVED** [review]

**Comment by amcasey** [comment]
The change LGTM but @andrewbranch might want to double-check the perf results.

**Comment by andrewbranch** [comment]
Looks good to me. A little bit of an increase but the absolute times are still very fast.

**Comment by bodinsamuel** [comment]
I'm sorry to ask, I know you probably have a lot on your hand. 
Is there an eta for merge? ☺️ (I don't know if you have planned release or it's just when you can)

**Comment by bodinsamuel** [comment]
@antoinerousseau thanks. Might be a merge issue, it was unintentional 🙇

**Pull request merged**